server/src/controllers/christmas-tree.es6

```diff
@@ -500,6 +497,43 @@ christmasTree.printPermit = (req, res) => {
     });
 };
 
+/**
+ * @function completePermitTransaction - method to take an update request to complete the permit
+ * and send complete transaction request to pay.gov and generate the permit email
+ * @param {Object} permit - object model of an existing permit to be completed
+ * @param {Object} req - http request
+ * @param {Object} res - http response
+ * @return {Object} - promise that the email has been sent and response too
+ */
+const completePermitTransaction = (permit, res, req) => {
+  util.logControllerAction(req, 'christmasTree.completePermitTransaction', permit);
+  const xmlData = paygov.getXmlToCompleteTransaction(permit.paygovToken);
+  return new Promise((resolve, reject) => {
+    paygov.postPayGov(xmlData)
+      .then(xmlResponse => {
+        return grabAndProcessTrackingId(res, xmlResponse, permit, req.body.status)
+          .then(updatedPermit => {
+            Promise.all([
+              returnSavedPermit(res, permit),
```

Comment:
if you make the above suggested change, then  `returnSavedPermit` can just be called normally, then handle the promise returned by `christmasTree.generateRulesAndEmail`